apps/ui/src/views/Space/Proposals.vue

```diff
@@ -38,24 +52,43 @@ function handleFetchVotingPower() {
 }
 
 watch(
-  [() => route.query.state as string],
-  ([toState]) => {
+  [
+    () => route.query.state as string,
+    () => route.query['labels[]'] as string[] | string
+  ],
+  ([toState, toLabels]) => {
     state.value = ['any', 'active', 'pending', 'closed'].includes(toState)
       ? (toState as NonNullable<ProposalsFilter['state']>)
       : 'any';
+    let normalizedLabels = toLabels || [];
+    normalizedLabels = Array.isArray(normalizedLabels)
+      ? normalizedLabels
+      : [normalizedLabels];
+    labels.value = normalizedLabels.filter(id => spaceLabels.value[id]);
+
     proposalsStore.reset(props.space.id, props.space.network);
-    proposalsStore.fetch(props.space.id, props.space.network, state.value);
+    proposalsStore.fetch(
+      props.space.id,
+      props.space.network,
+      state.value,
+      labels.value
+    );
   },
   { immediate: true }
 );
 
 watch(
-  [props.space, state],
-  ([toSpace, toState], [fromSpace, fromState]) => {
-    if (toSpace.id !== fromSpace?.id || toState !== fromState) {
+  [props.space, state, labels],
+  ([toSpace, toState, toLabels], [fromSpace, fromState, fromLabels]) => {
+    if (
+      toSpace.id !== fromSpace?.id ||
+      toState !== fromState ||
+      toLabels !== fromLabels
+    ) {
       const query: LocationQueryRaw = {
         ...route.query,
-        state: toState === 'any' ? undefined : toState
+        state: toState === 'any' ? undefined : toState,
+        'labels[]': !toLabels?.length ? undefined : toLabels
```

Comment:
i thought i added a comment in my previous review, but missed it,  can we make it just `labels` or `label` instead?
It looks like this now:
`http://127.0.0.1:8080/#/s:thanku.eth/proposals?labels[]=3fe1a9b9&labels[]=b4258576&labels[]=9ad0740d`
I think should be something like `labels=abc,xyz,etc`